brian2tools/mdexport/expander.py

```diff
@@ -0,0 +1,842 @@
+
+        spkgen : dict
+            Standard dictionary of SpikeGeneratorGroup
+        """
+        md_str = ''
+        md_str += (tab + 'Name ' + bold(spkgen['name']) +
+                ', with population size ' + bold(spkgen['N']) +
+                ', has neuron' + self.check_plural(spkgen['indices']) + ': ' +
+                ', '.join(str(i) for i in spkgen['indices']) +
+                ' that spike at times ' +
+                ', '.join(str(t) for t in spkgen['times']) +
+                ', with period ' + str(spkgen['period']) +
+                '.' + endll)
+        if 'run_regularly' in spkgen:
+            md_str += tab + bold('Run regularly: ') + endll
+            for run_reg in spkgen['run_regularly']:
+                md_str += self.expand_runregularly(run_reg)
+
+        return md_str
+
+    def expand_StateMonitor(self, statemon):
+        """
+        Expand StateMonitor from standard dictionary
+
+        Parameters
+        ----------
+
+        statemon : dict
+            Standard dictionary of StateMonitor
+        """
+        md_str = ''
+        md_str += (tab + 'Monitors variable' +
+                self.check_plural(statemon['variables']) + ': ' +
+                ','.join(
+                [self.render_expression(var) for var in statemon['variables']]
+                        ) +
+                ' of ' + statemon['source'])
+        if isinstance(statemon['record'], bool):
+            if statemon['record']:
+                md_str += ' for all members'
+        else:
+            # another bad hack (before with initializers)
+            if not statemon['record'].size:
+                md_str += ' for no member'
+            else:
+                md_str += (', for member' +
+                           self.check_plural(statemon['record']) +
+                           ': ' +
+                           ','.join([str(ind) for ind in statemon['record']]) +
+                           '.' + endll)
+        return md_str
+
+    def expand_SpikeMonitor(self, spikemon):
+        """
+        Expand SpikeMonitor from standard representation
+
+        Parameters
+        ----------
+
+        spikemon : dict
+            Standard dictionary of SpikeMonitor
+        """
+        return self.expand_EventMonitor(spikemon)
+
+    def expand_EventMonitor(self, eventmon):
+        """
+        Expand EventMonitor from standard representation
+
+        Parameters
+        ----------
+
+        eventmon : dict
+            Standard dictionary of EventMonitor
+        """
+        md_str = ''
+        md_str += (tab + 'Monitors variable' +
+                self.check_plural(eventmon['variables']) + ': ' +
+                ','.join(
+                [self.render_expression(var) for var in eventmon['variables']]
+                    ) +
+                ' of ' + eventmon['source'])
+        if isinstance(eventmon['record'], bool):
+            if eventmon['record']:
+                md_str += ' for all members'
```

Comment:
Not quite sure why, but I tested the `Diesmann_et_al_1999.py` example which uses a `SpikeMonitor` without any record argument (which therefore defaults to `record=True`), and the output is "for member: True" instead of "for all members", for some reason...